plugins/inputs/mysql/v2/convert.go

```diff
@@ -0,0 +1,94 @@
+package v2
+
+import (
+	"bytes"
+	"database/sql"
+	"fmt"
+	"strconv"
+)
+
+type ConversionFunc func(value sql.RawBytes) (interface{}, error)
+
+func ParseInt(value sql.RawBytes) (interface{}, error) {
+	v, err := strconv.ParseInt(string(value), 10, 64)
+
+	// Ignore ErrRange.  When this error is set the returned value is "the
+	// maximum magnitude integer of the appropriate bitSize and sign."
+	if err, ok := err.(*strconv.NumError); ok && err.Err == strconv.ErrRange {
+		return v, nil
+	}
+
+	return v, err
+}
+
+func ParseBoolAsInteger(value sql.RawBytes) (interface{}, error) {
+	if bytes.EqualFold(value, []byte("YES")) || bytes.EqualFold(value, []byte("ON")) {
+		return int64(1), nil
+	}
+
+	return int64(0), nil
+}
+
+func ParseGTIDMode(value sql.RawBytes) (interface{}, error) {
+	// https://dev.mysql.com/doc/refman/8.0/en/replication-mode-change-online-concepts.html
+	v := string(value)
+	switch v {
+	case "OFF":
+		return int64(0), nil
+	case "ON":
+		return int64(1), nil
+	case "OFF_PERMISSIVE":
+		return int64(0), nil
+	case "ON_PERMISSIVE":
```

Comment:
Is the permissive aspect of this mode not important?  I'm a little concerned that we're throwing away information.  Is that just for backwards compatibility?

Reply:
I did do this for backwards compatibility, but since we are converting to an integer we could add a 2/3 value.  This would mean we need to document the meaning of these values as they would be somewhat arbitrary.